code/reports/rmdchunks/score-forecasts.Rmd

```diff
@@ -0,0 +1,152 @@
+    ## "interval_score" first for this to work.
+    ## See https://github.com/epiforecasts/scoringutils/issues/111
+    metrics = c("interval_score", "coverage"),
+    compute_relative_skill = FALSE
+  ) %>%
+  filter(range %in% c(50, 95)) %>%
+  select(model, target_variable, horizon, location, coverage,
+                range) %>%
+  pivot_wider(
+           names_from = range, values_from = coverage,
+           names_prefix = "cov_"
+         )
+
+bias <- score_df %>%
+  filter(type != "point") %>%
+  eval_forecasts(
+    summarise_by = c("model", "target_variable", "horizon", "location"),
+    ## FIXME: we only care about coverage but we have to compute
+    ## "interval_score" first for this to work.
+    ## See https://github.com/epiforecasts/scoringutils/issues/111
+    metrics = c("interval_score", "bias"),
+    compute_relative_skill = FALSE
+  ) %>%
+  select(model, target_variable, horizon, location, bias)
+
+## relative absolute error of point forecast
+rel_ae <- score_df %>%
+  filter(type == "point", !is.na(true_value)) %>%
+  mutate(quantile = NA_real_) %>% ## scoringutils interprets these as point forecasts
+  eval_forecasts(
+    summarise_by = c(
+      "model", "target_variable",
+      "horizon", "location"
+    ),
+    compute_relative_skill = TRUE,
+    baseline = "EuroCOVIDhub-baseline",
+    rel_skill_metric = "ae_point"
+  ) %>%
+  select(model, target_variable, horizon, location, rel_ae = scaled_rel_skill)
+
+## for calculating WIS and bias, make sure all quantiles are there
+score_df <- score_df %>%
+  group_by(location, target_variable, target_end_date, model, horizon) %>%
+  mutate(all_quantiles_present =
+           (length(setdiff(quantiles, quantile)) == 0)) %>%
+  ungroup() %>%
+  filter(all_quantiles_present == TRUE) %>%
+  select(-all_quantiles_present)
+
+table <- score_df %>%
+  filter(type != "point") %>%
+  eval_forecasts(
+    summarise_by = c(
+      "model", "target_variable",
+      "horizon", "location"
+    ),
+    metrics = "interval_score",
+    compute_relative_skill = TRUE,
+    baseline = "EuroCOVIDhub-baseline"
+  ) %>%
+  select(-relative_skill) %>%
+  rename(rel_wis = scaled_rel_skill) %>%
+  full_join(rel_ae, by = c(
+                       "model", "target_variable", "horizon",
+                       "location"
+                     )) %>%
+  full_join(coverage, by = c(
+                        "model", "target_variable", "horizon",
+                        "location"
+                      )) %>%
+  full_join(bias, by = c(
+                    "model", "target_variable", "horizon",
+                    "location"
+                  )) %>%
+  left_join(num_loc, by = c(
+                       "model", "target_variable", "horizon",
+                       "location"
+                     )) %>%
+  left_join(num_fc, by = c(
+                      "model", "target_variable", "horizon",
+                      "location"
+                    )) %>%
+  left_join(locations, by = "location") %>%
```

Comment:
:grey_question: 

inor point: would it make sense to use the countrycode package here? It adds an extra dependency but it would make the intent much clearer IMO. At the beginning of this chunk, I was wondering why we need to save all the locations.

Reply:
I think that the country codes are also exported from covidHubUtils - so we could use those actually.

Reply:
I'm not sure either is a good idea. What if we want to run this on things that aren't the ECDC hub, or aren't countries (e.g. investigate our US forecasts that are state-by-state)? My inclination would be to make the chunks as self-contained as possible (i.e., get information from the existing tables if already present, rather than from outside packages), but perhaps I'm overemphasising the desire for this to be generalisable, at the expense of code stability/readability.

Reply:
Maybe the question is more genera then (and we might discuss it in another issue). Do we really need to have both `location_name` and `location` or can we avoid the pain of merging to get one from the other in many instances throughout the hub code?

Reply:
I'll answer my own comment: yes, we probably need both `location_name` and `location` in the case of other spatial cases. It's a bit clunky for us because there is no real ambiguity if we worked just the country names but if we think about regions or municipalities, it's much more robust and maybe even necessary to use codes.

Reply:
I'm not sure actually - it seems to me that `location_name` is superfluous but given that teams are asked to put it into the forecast submissions (and I think it's being validated) we may as well use it.